<a href="https://colab.research.google.com/github/eckoecho/Project-2/blob/main/World_Air_Quality_Index_by_City_and_Coordinates.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This data comes from [Kaggle, World Air Quality Index (AQI)](https://www.kaggle.com/datasets/adityaramachandran27/world-air-quality-index-by-city-and-coordinates).

The goal of this dataset is to provide valuable insights into the air quality of different regions, allowing researchers and policymakers to make informed decisions on how to address the issue of air pollution.


The target of our dataset is the `AQI Category` column. In this target I will determine if a country/city is a good choice to live based on the values of `Good`, `Moderate`, `Unhealthy for Sensitive Groups`, `Unhealthy`, and `Very Unhealthy`.

Each row represents a city within the world.

Our target is a categorical, deserving of a classification model for analysis.

There are 16695 rows, and 14 columns (13 features) in our dataset.

There are some challenges to learning this dataset including uncovering industry jargon and plotting geographical areas.

In [4]:
import pandas as pd

In [5]:
fpath="/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week07/Data/AQI and Lat Long of Countries.csv"
df = pd.read_csv(fpath)
df.head()

Country              City  AQI Value AQI Category  CO AQI Value  \
0  Russian Federation        Praskoveya         51     Moderate             1   
1              Brazil  Presidente Dutra         41         Good             1   
2              Brazil  Presidente Dutra         41         Good             1   
3               Italy   Priolo Gargallo         66     Moderate             1   
4              Poland         Przasnysz         34         Good             1   

  CO AQI Category  Ozone AQI Value Ozone AQI Category  NO2 AQI Value  \
0            Good               36               Good              0   
1            Good                5               Good              1   
2            Good                5               Good              1   
3            Good               39               Good              2   
4            Good               34               Good              0   

  NO2 AQI Category  PM2.5 AQI Value PM2.5 AQI Category      lat      lng  
0             Good               51           Moderate  44.7444  44.2031  
1             Good               41               Good  -5.2900 -44.4900  
2             Good               41               Good -11.2958 -41.9869  
3             Good               66           Moderate  37.1667  15.1833  
4             Good               20               Good  53.0167  20.8833

In [6]:
#indicating rows and columms
df.shape
print(f"There are {df.shape[0]} rows, and {df.shape[1]} columns in our dataset.")

There are 16695 rows, and 14 columns in our dataset.


In [7]:
#A quick glance at our features
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16695 entries, 0 to 16694
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Country             16393 non-null  object 
 1   City                16695 non-null  object 
 2   AQI Value           16695 non-null  int64  
 3   AQI Category        16695 non-null  object 
 4   CO AQI Value        16695 non-null  int64  
 5   CO AQI Category     16695 non-null  object 
 6   Ozone AQI Value     16695 non-null  int64  
 7   Ozone AQI Category  16695 non-null  object 
 8   NO2 AQI Value       16695 non-null  int64  
 9   NO2 AQI Category    16695 non-null  object 
 10  PM2.5 AQI Value     16695 non-null  int64  
 11  PM2.5 AQI Category  16695 non-null  object 
 12  lat                 16695 non-null  float64
 13  lng                 16695 non-null  float64
dtypes: float64(2), int64(5), object(7)
memory usage: 1.8+ MB


In [8]:
#Viewing the spectrum of values
df.describe()

AQI Value  CO AQI Value  Ozone AQI Value  NO2 AQI Value  \
count  16695.000000  16695.000000     16695.000000   16695.000000   
mean      62.998682      1.342138        31.767355       3.819647   
std       43.091971      2.371379        22.839343       5.880677   
min        7.000000      0.000000         0.000000       0.000000   
25%       38.500000      1.000000        20.000000       0.000000   
50%       52.000000      1.000000        29.000000       2.000000   
75%       69.000000      1.000000        38.000000       5.000000   
max      500.000000    133.000000       222.000000      91.000000   

       PM2.5 AQI Value           lat           lng  
count     16695.000000  16695.000000  16695.000000  
mean         59.821324     30.267148     -3.944485  
std          43.208298     22.947398     73.037148  
min           0.000000    -54.801900   -171.750000  
25%          34.000000     16.515450    -75.180000  
50%          52.000000     38.815800      5.643100  
75%          69.000000     46.683300     36.275000  
max         500.000000     70.767000    178.017800

> Note: An AQI value of 0.0000 for ozone may occur if the ozone concentration in the air is extremely low or falls below the detection limit of the monitoring equipment. It does not necessarily mean that ozone is completely absent, but rather that it is present at a very low concentration that falls below the threshold for reporting.

# Clean the Data

In [9]:
#Looking for duplicates to drop
df.duplicated().sum()

0

No duplicate values to drop.

In [30]:
#Looping thru the dataset's column values to find any inconsistencies
#I'm including dropna=False to see any NaN values
for col in df:
    print(f"{df[col].value_counts(dropna=False)} \n\n")

United States of America    3954
Germany                     1072
Brazil                       890
Russian Federation           889
Italy                        866
                            ... 
Andorra                        1
Luxembourg                     1
Saint Lucia                    1
Guinea-Bissau                  1
Monaco                         1
Name: Country, Length: 175, dtype: int64 


Santa Cruz      17
San Fernando    16
Santa Ana       15
San Juan        14
Springfield     12
                ..
Hegang           1
Herxheim         1
Onalaska         1
Ostfildern       1
Westerville      1
Name: City, Length: 14229, dtype: int64 


50     413
52     374
35     366
51     359
54     352
      ... 
234      1
232      1
225      1
283      1
252      1
Name: AQI Value, Length: 282, dtype: int64 


Good                              7708
Moderate                          7054
Unhealthy                          871
Unhealthy for Sensitive Groups     869
Very Unhealthy    

In [28]:
#Checking for missing values
df.isna().sum()

Country               302
City                    0
AQI Value               0
AQI Category            0
CO AQI Value            0
CO AQI Category         0
Ozone AQI Value         0
Ozone AQI Category      0
NO2 AQI Value           0
NO2 AQI Category        0
PM2.5 AQI Value         0
PM2.5 AQI Category      0
lat                     0
lng                     0
dtype: int64

The `Country` feature has 302 missing values.  I'm going to look at the most common value for this column. However, I may be able to deduce the Country for `lat` and `lng` features.

In [29]:
#Looking for the most common Country in our dataset
df["Country"].value_counts()

United States of America    3954
Germany                     1072
Brazil                       890
Russian Federation           889
Italy                        866
                            ... 
Luxembourg                     1
Saint Lucia                    1
Republic of Korea              1
Solomon Islands                1
Monaco                         1
Name: Country, Length: 174, dtype: int64